### PR TITLE
Derek defaults codegen

### PR DESCRIPTION
**Pull request by xiexr151e**

### CODE REVIEW

File: compiler.ts [hunk 1]
```diff
@@ -68,7 +69,7 @@ export function makeLocals(locals: Set<string>) : Array<string> {
 
 }
 
-export function compile(ast: Program<Type>, env: GlobalEnv) : CompileResult {
+export function compile(ast: Program<Type>, env: GlobalEnv, tenv: GlobalTypeEnv) : CompileResult {
```
Comment (DavidLiu0304) [inline compiler.ts:72]: I think type environment should be used in type-checker.ts

**Review by danielytan — COMMENTED** [review]
Looks good, but this code may be redunant (i.e. the typechecker already adds default args when necessary.)

**Review by DavidLiu0304 — COMMENTED** [review]
Some parts of this code change is redundant and handled in the typechecker

**Comment by danielytan** [comment]
We appreciate the work, but passing around the type environment in the compiler may not be necessary, and the typechecker seems to already sufficiently handle default args.

**Pull request closed without merging**